Support installation via Homebrew

Dschee opened the pull request:
Please note that for this to actually work, you will need to do the following steps:

1. Make a new tagged release (must be committed before next step)
2. Update the `Formula/swiftinfo.rb` with the correct tagged commit and its commit hash

Then, this should actually work. I have already tested this by making tagged releases (e.g. `50.50.50`) in my fork and pointing the Formula to my fork, it worked with the changes in e7050a6. (Basically the commit fixes issues with symlinks, cause Homebrew only adds a symlink to the final executable path.)

Note that you can now install the tool locally by running `make install` and uninstall it via `make uninstall`. Also I added a `make portable_zip` command which you may need to insert a `copy` command into, then you should even be able to remove the `prepare_for_release.sh` script. But I leave that choice to you. @rockbruno 

This fixes #17.

rockbruno commented:
Awesome! Thanks for the help!
I had committed the tarball and formula to try it out, but I wasn't able to make it work because I couldn't make it find the module map files due to the symlinks, and I didn't wanted to copy it to `bin`.
Do you think it's possible to copy `Csourcekitd` to the formula's `include` folder and use that instead?

Dschee commented:
I'm not sure what you mean, but I solved the symlink issue for installation via Hombebrew in e7050a6d13ed59404e69c4bf71d7ed84c0dbf2b4.

Do you mean for when you make a tarball release? I'm not experienced with such a thing, I've only copied over the `portable_zip` make commands from my other project [BartyCrouch](https://github.com/Flinesoft/BartyCrouch/blob/stable/Makefile#L28) where someone contributed the code. It's working, but I'm not an expert on that. Sorry. 😓

rockbruno commented:
I mean that it's working because you're copying `Csourcekitd` to `/usr/local/bin` together with the binary, right? I was worried about doing this because it could conflict with something else (for example, sourcekit-lsp also has a folder called `Csourcekitd`), and I think this makes homebrew unable to uninstall it as well

In the example that I was following, the "correct" seems to be copying everything to homebrew's internal folders and symlinking it back to `/usr/bin`:

```
  def install
    bin.install "swiftinfo"
    include.install Dir["*"]
  end
```

This will move the files from the tarball to the internal folders but link the binary to `/usr/bin` and the rest to `/usr/include`. I just couldn't get it to run because it was trying to find the module map in `/usr/bin` instead of the binary's real folder, but I think the changes you made to `toolFolder` might actually fix this!

Dschee commented:
No, I'm not copying to `/usr/local/bin` but instead to `/usr/loca/Cellar/swiftinfo/2.2.0/bin`. I think that's where the confusion comes from. :) In `/usr/local/bin` there's only a symlink which I follow to the `Cellar` folder, where the `Csourcekitd` files reside.

rockbruno commented:
I got it to work with your changes! Give me a second, my idea is to merge this into a temp branch so I can open another PR with some changes from my side (I'm changing the folder structure so we can ship the pods + homebrew with a single .zip)

rockbruno commented:
Want to add a changelog entry before I merge? 🎉

Dschee commented:
Good catch, I just added a changelog entry. Please note that I added links to both GitHub profiles and related issues / PRs. I'm not sure what exactly you use the CHANGELOG for, but if it's used in GitHub releases, having links always helps learning more about a feature if any questions arise.

rockbruno commented:
I use it just for community purposes, but I like what you did! 👍
